tests/unit/main.fmf

```diff
@@ -1,10 +1,78 @@
 summary: Python unit tests
 description:
     Run all available python unit tests using pytest.
-test: python3 -m pytest -vvv -ra --showlocals
-framework: shell
-require:
-  - python3-pytest
+duration: 30m
 environment:
     LANG: en_US.UTF-8
-tier: 0
+
+    ENABLE_PARALLELIZATION: "no"
+    ENABLE_CONTAINERS: "no"
+    WITH_SYSTEM_PACKAGES: "no"
+
+adjust+:
+  - when: distro == centos-stream-9
+    environment+:
+      HATCH_ENVIRONMENT: dev-not-editable
+
+require+:
+  - gcc
+  - git
+  - python3-nitrate
+  - libvirt-devel
+  - krb5-devel
+  - libpq-devel
+  - python3-devel
+  - jq
+  - podman
+  - buildah
+  - hatch
+
+# Run against development packages via `hatch`.
+/with-development-packages:
+    enabled: false
+
+    adjust+:
+      - when: initiator is not defined or distro == fedora-39
+        because: Enable locally or in CI on Fedora 39
```

Comment:
Let's not hard code `fedora-39` here. The limitation to `f39` is already covered by the `packit` definition.

Reply:
Hmm, so what should I leave here? If I drop the `distro` check, the test will not run anywhere and remain disabled. If I enable it by default, it will run on all distros. Something must tell the test to run on a particular distro only. I don't see how Packit would enforce it, it runs `/plans/features/core` which includes `/tests/unit/with-development-packages/basic`, and the test would on all distros. The `/extended` variant would be skipped because it's excluded from `/plan/features/core`, it would indeed run in its own Packit check, but `/basic` does not have a dedicated Packit job.

Reply:
Oh, I see, I mixed this with the extra plan which has dedicated `packit` jobs. Hmmm... so what about using `fedora-rawhide` here? So that we don't have to rewrite/update the value in the future?

Reply:
Ah, I wanted to avoid that :) There seems to be some misunderstanding between `podman` and `pytest-container`, which makes the extended set crash and burn. I was hoping to use some older Fedora for the moment and leave the investigation for later. Would that be acceptable?

Reply:
Understood. Sure. Let's ignore this then.

Reply:
Would be nice to use packit aliases for this. I remember there was an announcement of a self-contained project that exposes such aliases.

Reply:
I'd say so, yes. fmf recognizing and/or exposing `fedora-latest-stable`, for example.

Reply:
We could ask `packit` to provide the `fedora-latest-stable` alias as well:

    distro = ['fedora-40', 'fedora-latest-stable']

This would allow matching the `adjust` rules. Would that make sense? Could be dangerous, for example:

    when: distro != fedora-40

This would match `fedora-latest-stable` and that is not expected. Putting this into `fmf` does not probably make sense either, `fmf` is dumb about the `context` dimensions. There is not special handling for `distro`.

Reply:
Anyway, let's track this separately.

Reply:
Note: similar problem as https://github.com/teemtee/tmt/issues/472